pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -20,16 +20,30 @@ var (
 	getVulnerabilityDetailFunc = db.Config{}.GetVulnerabilityDetail
 )
 
-func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, string, string, []string) {
+func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, types.VendorVectors, string, string, []string) {
 	details, err := getVulnerabilityDetailFunc(vulnID)
 	if err != nil {
 		log.Println(err)
-		return types.SeverityUnknown, nil, "", "", nil
+		return types.SeverityUnknown, nil, nil, "", "", nil
 	} else if len(details) == 0 {
-		return types.SeverityUnknown, nil, "", "", nil
+		return types.SeverityUnknown, nil, nil, "", "", nil
 	}
 
-	return getSeverity(details), getVendorSeverity(details), getTitle(details), getDescription(details), getReferences(details)
+	return getSeverity(details), getVendorSeverity(details), getVendorVectors(details), getTitle(details), getDescription(details), getReferences(details)
+}
+
+func getVendorVectors(details map[string]types.VulnerabilityDetail) types.VendorVectors {
+	vv := make(types.VendorVectors)
+	for vendor, detail := range details {
+		if detail.CvssVector == "" && detail.CvssVectorV3 == "" {
+			continue
+		}
+		vv[vendor] = types.CVSSVector{
+			V2: detail.CvssVector,
+			V3: detail.CvssVectorV3,
+		}
```

Comment:
info: If we re-use CVSSVector, we can write something like the following.

```suggestion
		vv[vendor] = detail.CvssVector
```

But, it's no big deal 😉

Reply:
I tried your suggestion but I would suggest let's keep it this way. After trying, I like the analog between the V2 and V3 by having a field for each.